### ISSUE TITLE
Property Indexing Mode - Enhancements and Bugs

### ISSUE THREAD

**Issue by devinjackson** [tsgrp/hpi#883]
Below is a list of enhancements we would like to add to the property indexing mode in blukupload:

Enhancements: 
1. Text areas and any other input type we want need to be able to be filled, not just input boxes.
2. The height of the indexing iFrame can be adjusted dynamically.
3. This only works with alfresco right now. We need to do some extra work to get it to work with  Hbase and documentum.
6. Possibly reduce the amount of highlighting necessary to pull data from a document.
7. Clean up indexing the temporary files automatically
8. If we can't generate a pdf of a document, we need to skip it when filling in properties.

Bugs:
1. Multi-property selecting (click on one property, change to a different one, both properties get filled).
2. When you click the "Next" button the first time on the bulkproperties screen, the iFrame appears.  As of right now, you have to click the file name at the top of the bulkproperties view to get the iFrame to appear the first time.

**Comment by m-brouillette** [comment]
For Enhancement #6 , OpenAnnotate can be configured to have a lower snapThreshold.  https://github.com/tsgrp/OpenAnnotate/wiki/OA-Configuration-Files#snapthreshold

The one warning would be that snapThreshold's lower than 0.1 (10%) haven't been tested much and could have issues in OA, so we may not be at "Click a word to highlight it," just yet.

**Comment by gsteimer** [comment]
@m-brouillette @benallenallen - are all of these truly Upon Client Request?  Or are there some that we should definitely implement before a client goes into production with Property Indexing Mode enabled?

**Comment by benallenallen** [comment]
I would agree that "Issues" #1 and #2 need to be fixed before we call this ready for client consumption.

Enhancement #7 is also a fairly high priority item before we say it is production ready.
Enhancements #1-6 seem like fine enhancements for a client to pay for if they don't like how it behaves (or that it doesn't at all work on DCTM yet)

**Comment by m-brouillette** [comment]
I agree with Ben, Issues 1 & 2 need to be fixed, and Enhancement 7 should be priorities if HUD indexing is going to be used anywhere other than demos.

**Comment by gsteimer** [comment]
@benallenallen @m-brouillette @devinjackson - I updated the original request based on the discussion above.  Please verify.  Also - checkboxes work better to track what's been completed vs. numbering the items.

**Comment by aviking** [comment]
The Issues section of this issue were fixed with HPI Commit: 14094 and OA Commit: 2930. Both Code Reviewed by Ben A. Need to Write unit tests for the updates to Bulk Upload and Form Support (see HPI commit: 13619 and 14094 to see parts of code that were updated to figure out what unit tests need to be written. Talk to @aviking if you need assistance figuring out what unit tests need to be written.
